From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 15 Jan 2019 14:57:25 -0500
Subject: [PATCH] Try to set -fPIE and friends on libgnu.a

In order to make sure UTIL_CFLAGS and UTIL_LDFLAGS can correctly get
-Wl,-z,relro,-z,now , we need everything going in them to be built with at
least -fPIC (and preferably -fPIE) wherever we can, or else we get relocations
in some component object that can't be used with the link type that's being
used for the final ELF object.

So this makes sure libgnu.a gets built with HOST_CFLAGS and HOST_LDFLAGS,
which are what is later used to define UTIL_CFLAGS and UTIL_LDFLAGS, and
includes -fPIE.

Fixes an rpmdiff check.

Related: rhbz#1658500

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 grub-core/gnulib/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/grub-core/gnulib/Makefile.am b/grub-core/gnulib/Makefile.am
index b7c5e60e1c3..bd3621930ff 100644
--- a/grub-core/gnulib/Makefile.am
+++ b/grub-core/gnulib/Makefile.am
@@ -38,8 +38,8 @@ CLEANFILES =
 DISTCLEANFILES =
 MAINTAINERCLEANFILES =
 
-AM_CPPFLAGS =
-AM_CFLAGS =
+AM_CPPFLAGS = $(HOST_CPPFLAGS)
+AM_CFLAGS = $(HOST_CFLAGS)
 
 noinst_LIBRARIES += libgnu.a
 
